src/values/AbstractValue.js

```diff
@@ -293,6 +320,24 @@ export default class AbstractValue extends Value {
   }
 
   refineWithPathCondition(): Value {
+    if (this.kind === "&&") {
+      let [left, right] = this.args;
+      let refinedLeft = left instanceof AbstractValue ? left.refineWithPathCondition() : left;
+      let refinedRight = right instanceof AbstractValue ? right.refineWithPathCondition() : right;
+      if (!refinedLeft.mightNotBeTrue()) return refinedRight;
```

Comment:
This logic here seems more general that the place where it sits: `createFromLogicalOp` could do this kind of normalization to benefit anyone who attempts to build logical operations.

Reply:
I don't want the abstract value factory functions to be path sensitive. There is probably room for an in depth design discussion about all this, but it is a bit premature right now.

Reply:
Totally agreed that the factory should not be path sensitive. I just meant the mightNotBeTrue() kind of checks that result in simplified expressions.